### PR TITLE
Multi account support in repo tool

### PR DESCRIPTION
**Pull request by dhoehna**
## Summary of the pull request
This PR adds multi account support to the repo tool  This includes
1. Logging into a different account
2. Selecting repos from both accounts
3. Switching between accounts.

## Detailed description of the pull request / Additional comments

This PR has other changes that aren't related to multi-account support.
1. FolderPickerViewModel moved from AddRepoDialog.xaml.cs to AddRepoViewModel.cs.  This is part of the effort to move all code from code-behind to the view model.
2. Filtering in the repo tool now uses .contains() instead of .starts with.

## Validation steps performed
Manual tests.

## PR checklist
- [ ] Closes #xxx
- [ X] Tests added/passed
- [ ] Documentation updated

Movie!
Logging into an account via the repo tool
LoggingInWithRepoTool

Cloning repos after logging in!
CloningAfterAddingAnAccount

Pictures!
The X button to cancel the login prompt.
LoginWithXButton

### CODE REVIEW

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs [hunk 1]
```diff
@@ -298,12 +328,127 @@ private void CancelButtonPressed()
...
+
+            ShowRepoPage = Visibility.Collapsed;
+
+            // Store the logged in accounts to help figure out what account the user logged into.
+            var loggedInAccounts = await Task.Run(() => _providers.GetAllLoggedInAccounts(_selectedRepoProvider));
+            await LogUserIn(_selectedRepoProvider, _addRepoDialog.GetLoginUiContent(), true);
+            var loggedInAccountsWithNewAccount = await Task.Run(() => _providers.GetAllLoggedInAccounts(_selectedRepoProvider));
+
+            ShowRepoPage = Visibility.Visible;
+            Accounts = new ObservableCollection<string>(loggedInAccountsWithNewAccount.Select(x => x.LoginId));
```
Comment (vineeththomasalex) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs:413]: Check for x.Url which is unique (since GHES breaks uniqueness of LoginId)

**Review by dkbennett — APPROVED** [review]

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs (hunk 1 above)
Comment (dhoehna) [reply tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs:413]: I'm going to leave this as is for now.  Changing everything to support .Url is too much for this PR.

**Comment by dhoehna** [comment]
Pipelines are bad.  Moving to a new PR.  #2140

**Pull request closed without merging**